xc/device.ml

```diff
@@ -2336,19 +2353,35 @@ module Backend = struct
         (* Take the first 'max_emulated_nics' elements from the list. *)
         let nics = Xapi_stdext_std.Listext.List.take Dm_Common.max_emulated_nics nics in
 
-        let nics' =
-          if nics <> [] then
-            List.map (fun (mac, bridge, devid) ->
-                [
-                  "-device"; sprintf "rtl8139,netdev=tapnet%d,mac=%s,addr=%d" devid mac (devid + 4);
-                  "-netdev"; sprintf "tap,id=tapnet%d,ifname=tap%d.%d,script=no,downscript=no" devid domid devid
-                ]
-              )
-              nics
-          else
-            [["-net"; "none"]]
-        in
-        common @ misc @ (List.concat nics')
+        (* add_nic is used in a fold: it adds fd and command line args
+         * for a nic to the existing fds and arguments (fds, argv)
+         *)
+        let none = ["-net"; "none"] in
+        let add_nic (fds, argv) (mac, bridge, devid) =
+          let ifname          = sprintf "tap%d.%d" domid devid in
+          let uuid, fd as tap = tap_open ifname in
```

Comment:
`fd` is unused, is that expected? In that case let's ignore it with `_` or `_fd`